nebula-logger/core/main/logger-engine/classes/LoggerSObjectHandler.cls

```diff
@@ -0,0 +1,283 @@
+    }
+
+    protected virtual void executeAfterUpdate(Map<Id, SObject> triggerNewMap, Map<Id, SObject> triggerOldMap) {
+    }
+
+    protected virtual void executeAfterDelete(Map<Id, SObject> triggerNewMap) {
+    }
+
+    protected virtual void executeAfterUndelete(Map<Id, SObject> triggerNewMap) {
+    }
+
+    private void executePlugins() {
+        if (this.pluginConfigurations == null || this.pluginConfigurations.isEmpty() == true) {
+            return;
+        }
+
+        for (LoggerPlugin__mdt pluginConfiguration : this.pluginConfigurations) {
+            if (String.isNotBlank(pluginConfiguration.SObjectHandlerApexClass__c)) {
+                this.executeApexPlugin(pluginConfiguration);
+            }
+            if (String.isNotBlank(pluginConfiguration.SObjectHandlerFlowName__c)) {
+                this.executeFlowPlugin(pluginConfiguration);
+            }
+        }
+    }
+
+    @SuppressWarnings('PMD.AvoidDebugStatements')
+    private void executeApexPlugin(LoggerPlugin__mdt configuration) {
+        try {
+            LoggerPlugin.Triggerable apexPlugin = LoggerPlugin.newTriggerableInstance(configuration.SObjectHandlerApexClass__c);
+            apexPlugin?.execute(configuration, this.input);
+
+            if (System.Test.isRunningTest() == true && apexPlugin != null) {
+                this.executedApexPlugins.add(apexPlugin);
+            }
+        } catch (TypeException ex) {
+            System.debug(LoggingLevel.WARN, 'Unknown Apex class ' + configuration.SObjectHandlerApexClass__c + ', skipping plugin execution');
+        }
+    }
+
+    @SuppressWarnings('PMD.AvoidDebugStatements')
+    private void executeFlowPlugin(LoggerPlugin__mdt configuration) {
+        try {
+            Map<String, Object> flowInputs = new Map<String, Object>();
+            flowInputs.put('pluginConfiguration', configuration);
+            flowInputs.put('pluginInput', this.input);
+
+            Flow.Interview flowPlugin = Flow.Interview.createInterview(configuration.SObjectHandlerFlowName__c, flowInputs);
+            flowPlugin.start();
+
+            List<SObject> updatedTriggerNew = (List<SObject>) flowPlugin.getVariableValue('updatedTriggerNew');
+            if (updatedTriggerNew != null && updatedTriggerNew.size() == this.input.triggerNew.size()) {
+                this.input.triggerNew.clear();
+                this.input.triggerNew.addAll(updatedTriggerNew);
+                if (this.input.triggerNewMap != null) {
+                    this.input.triggerNewMap.clear();
+                    this.input.triggerNewMap.putAll(new Map<Id, SObject>(updatedTriggerNew));
+                }
+            }
+            if (System.Test.isRunningTest() == true && flowPlugin != null) {
+                this.executedFlowPlugins.add(flowPlugin);
+            }
+        } catch (TypeException ex) {
+            System.debug(LoggingLevel.WARN, 'Unknown Flow ' + configuration.SObjectHandlerFlowName__c + ', skipping plugin execution');
```

Comment:
might there be some value in using `Logger.warn` here? (and the other instances of `System.debug` used in this class? That way you can pass your custom message _and_ the caught exception 😃

Reply:
Yeah, sounds good to me - not sure why I didn't do that originally, I even have [a custom metadata record, LoggerParameter.EnableLoggerSystemMessages](https://github.com/jongpie/NebulaLogger/blob/main/nebula-logger/core/main/configuration/customMetadata/LoggerParameter.EnableLoggerSystemMessages.md-meta.xml) that I use for this situations where Logger will create system entries (so orgs can disable it if they don't want the entries generated). I'll update all of these `System.debug()` instances!

Reply:
yeah, I guess so long as orgs have a way to opt-out of those `Logger` specific entries being created, it would be nice to dogfood the log messages here